### PR TITLE
v0.5.0

### PR DESCRIPTION
**Pull request by superkat64**
### Description

Minor Version Update. No breaking changes.
Bringing SignalerJS into modern age of javascript. 

### Changes
Updated most dependencies. Revamped the test suite from Karma to Jest. Updated configuration for Babel, esLint, Webpack, Node, and removed all unnecessary dependencies. 

###### Removed
- `babel-preset-es2015`
- `chai`
- `istanbul-instrumenter-loader`
- `karma`
- `karma-chai`
- `karma-cli`
- `karma-coverage`
- `karma-mocha`
- `karma-phantomjs-launcher`
- `karma-sourcemap-loader`
- `karma-webpack`
- `mocha`
- `mocha-loader`
- `phantomjs`
- `sinon`
- `whatwg-fetch`

###### Added
- `"@babel/preset-env": "^7.0.0",
- `"eslint-config-prettier": "^8.3.0"`
- `"eslint-plugin-jest": "^26.0.0"`
- `"jest": "^27.4.7"`
- `"prettier": "2.5.1"`

###### Updated
- `"@babel/core": "^7.0.0"`
- `"@babel/eslint-parser": "^7.0.0"`
- `"@babel/eslint-plugin": "^7.0.0"`
- `"es6-promise": "4.2.4"`
- `"eslint": "^8.0"`
- `"webpack": "^5.0"`
- `"blueimp-md5": "2.19.0"`
- `"cookies-js": "1.2.3"`

### Ticket

[PEG-30](https://customink.atlassian.net/browse/PEG-30)

### Screenshots

Screen Shot 2022-02-01 at 12 28 41 PM

### Notes

_Recommended reading: [Code Review guide](https://github.com/customink/guides/blob/master/operations/code-review/README.md)_

### Optional Tasks

- [ ] Include 🎩 Instructions
- [ ] Update the readme (README.md)
- [ ] Update the API or architecture docs (e.g. docs/api.md)

##### Library-Specific

- [x] Increment the changelog (CHANGELOG.md)
- [x] Increment the version number (lib/version.rb)
- [ ] [Release & Tag][release] the version above in Github

[release]: https://docs.github.com/en/github/administering-a-repository/managing-releases-in-a-repository

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+node_modules
+coverage
+npm-debug.log
+signalerjs.min.js
+package-lock.json
```
Comment (avinash-vllbh) [inline .prettierignore]: Nit: Missing new line.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+- Added [Prettier](https://prettier.io/)
+
+### Changed
+
+- Updated the outdated Karma tests to Jest tests
+- Bumped Node, Babel, Webpack, & ESLint versions
+
+### Removed
+
+- All other testing libraries: karma, chai, sinon, phantomjs, mocha, istanbul
```
Comment (avinash-vllbh) [inline CHANGELOG.md:25]: ❤️ the changelog. Thanks for adding it.

File: .prettierrc.js [hunk 3]
```diff
@@ -0,0 +1 @@
+module.exports = {};
```
Comment (avinash-vllbh) [inline .prettierrc.js]: We have a standard prettier config that most of other apps use - e.g https://github.com/customink/ci-spa-starter/blob/main/templates/react-query/.prettierrc.js

**Review by avinash-vllbh — APPROVED** [review]
Left a few minor comments. 💯 overall.

File: .prettierrc.js (hunk 3 above)
Comment (superkat64) [reply .prettierrc.js]: Just a heads up: 
got this warning while running `npx prettier --write .` with the config rules applied
```
[warn] jsxBracketSameLine is deprecated.
```

**Pull request merged**